src/mscorlib/shared/System/Number.NumberBuffer.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class Number
+    {
+        // Constants used by number parsing
+        private const Int32 NumberMaxDigits = 32; // needs to == NUMBER_MAXDIGITS in coreclr's src/classlibnative/bcltype/number.h
```

Comment:
The "NumberMaxDigits" in CoreRT was actually 31 (with the buffer size being 32 to allow for the NUL terminator.)

Reply:
Hmm, it's seems that it's actually 32 in CoreRT but CoreRT is missing the extra space for the NUL terminator.

Reply:
Right now I have it as 32 digits + space for the null terminator. Should I make it 31+1 instead? Or fine as is?

Reply:
Might have to make it 50+1 after all if we want to be fully compatible: the 50 seems to have come from the fact that the `NumberToDecimal` looks ahead 20 digits to decide whether to round up or down when you have a string at the precision threshold ending with an even number followed by a '5':

https://github.com/dotnet/coreclr/blob/d3297abfafcf8f89de5816145afc955b0710c051/src/classlibnative/bcltype/decimal.cpp#L353

So 29 digits for Decimal max precision + 20 digits lookahead + 1 to sleep better at night = 50.

Consider this example:

```
StringBuilder sb = new StringBuilder();
sb.Append('.');
sb.Append('2', 28);
sb.Append('5');
sb.Append('0', 10);
sb.Append('1');
String s = sb.ToString();

Decimal d = Decimal.Parse(s);
Console.WriteLine(d);
```

prints:

   0.2222222222222222222222222223  

on desktop and

   0.2222222222222222222222222222

on ProjectN

Personally, I'd be fine with either 50+1 or 32+1 - the "20 digits lookahead" is a pretty arbitrary policy. But we should make a decision and make it match (Utf8Parser as well.)

Reply:
Ok, to minimize impact on coreclr here I'll stick with 50+1.  We can subsequently change to 32+1 if desired.